Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/QuadrigaAccessDeniedHandler.java

```diff
@@ -0,0 +1,35 @@
+package edu.asu.spring.quadriga.web.config;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.access.AccessDeniedHandler;
+
+/**
+ * This handler is invoked when users try to access a protected URL without the necessary 
+ * access rights to a page
+ * 
+ * @author Chiraag Subramanian
+ *
+ */
+
+public class QuadrigaAccessDeniedHandler implements AccessDeniedHandler {
+    private final Logger logger = LoggerFactory.getLogger(QuadrigaAccessDeniedHandler.class);
+    
+    @Override
+    public void handle( HttpServletRequest request, HttpServletResponse response, AccessDeniedException exc) throws IOException, ServletException {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        if (auth != null) {
+            logger.warn("User: " + auth.getName() + " attempted to access the protected URL: " + request.getRequestURI());
```

Comment:
let's not use the warn level here. This might easily happen when a user is logged out after their session expired, and they try to continue working. This isn't worth a warning. Make it info or debug.